internal/health/condition/check_all_members_updated.go

```diff
@@ -0,0 +1,59 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package condition
+
+import (
+	"context"
+	"fmt"
+
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
+	"github.com/gardener/etcd-druid/internal/utils"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type allMembersUpdated struct {
+	cl client.Client
+}
+
+func (a *allMembersUpdated) Check(ctx context.Context, etcd druidv1alpha1.Etcd) Result {
+	res := &result{
+		conType: druidv1alpha1.ConditionTypeAllMembersUpdated,
+	}
+
+	sts, err := utils.GetStatefulSet(ctx, a.cl, &etcd)
+	if sts == nil && err == nil {
```

Comment:
just reverse it and it will be simplified.
```go
if err != nil {
  res.status = druidv1alpha1.ConditionUnknown
  res.reason = "UnableToFetchStatefulSet"
  res.message = fmt.Sprintf("Unable to fetch StatefulSet for etcd: %s", err.Error())
  return res
} else if sts == nil {
  res.status = druidv1alpha1.ConditionUnknown
  res.reason = "StatefulSetNotFound"
  res.message = fmt.Sprintf("StatefulSet %s not found for etcd", etcd.Name)
  return res
}
```

Reply:
Thanks for the catch. I had copied over the base code from `check_data_volumes_ready.go` to make things easy, and didn't seem to notice this. I have now fixed that one as well.